### PR TITLE
add function specializations for the case of parameters in BFloat16 data type

### PR DESCRIPTION
**Pull request by CaoE**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100234
* __->__ #100233

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/cpu/group_norm_kernel.cpp [hunk 1]
```diff
@@ -1367,7 +1542,7 @@ void GroupNormBackwardKernelImpl(
   // It will keep module parameters in acc dtype i.e. float
   // while input/output will be in BFloat16.
   // Using parameters in BFloat16 will cause high precision loss.
-  const bool mixed_type = is_mixed_type(dY, gamma);
+  const bool mixed_type = is_mixed_type(dY, mean);
```
Comment (jgong5) [inline aten/src/ATen/native/cpu/group_norm_kernel.cpp:1545]: The comment doesn't correlate well with the code any more? Can you revise the comment?

**Review by jgong5 — COMMENTED** [review]
How about half?

File: aten/src/ATen/native/cpu/group_norm_kernel.cpp (hunk 1 above)
Comment (CaoE) [reply aten/src/ATen/native/cpu/group_norm_kernel.cpp:1545]: Sure, revised at https://github.com/pytorch/pytorch/pull/100234.

**Review by jgong5 — APPROVED** [review]

**Comment by CaoE** [comment]
@malfet Could you please help to review this PR ? Thanks.

**Comment by CaoE** [comment]
@ngimel  Could you please help to review this PR ? Thank you.

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**